### DISCUSSION TITLE
PG Database - selecting schemas

### DISCUSSION THREAD

**Discussion by akrea** [nocodb/nocodb#6809] [Q&A]
Hi 

Already asked my question [here](https://github.com/nocodb/nocodb/issues/226), but since it is a closed issue, I thought nobody would actually see it. For you not needing to use the link I'll repeat the question here:

In the latest revision of the UI it is not possible (or at least I did not figure out how) to connect to a specific schema. Can you help me out with this?

As a further note: We do integrate nocodedb in an existing database. nocodb writes its core data into the public schema. It would be really helpful to be able to tell nocodb to use a specific nocodb-Schema so that the public schema remains pristine for other project related table that really need to be in public schema. Or is it already possible?

**Comment by o1lab** [comment]
While connecting to external database with a postgres schema other than public - you can use the `schema` field as shown below (last input field)

<img width="622" alt="image" src="https://github.com/nocodb/nocodb/assets/5435402/1eb7d2ad-6bcf-4532-8d69-c3d37115ed9f">

Coming to pointing our core tables to a different schema - you can use NC_DB* related env variables. I would suggest you point it to a new postgres db (in same physical server). Read about NocoDB's env variables [here](https://docs.nocodb.com/getting-started/self-hosted/environment-variables/).

<img width="1306" alt="image" src="https://github.com/nocodb/nocodb/assets/5435402/c476d979-b4fb-42f8-8ce4-387d5e95044b">

**Reply by akrea** [reply]
Thank you very much for the quick reply. The thing is: when I add an "external" data source, the auditing function does not work (or at least I did not see how to activate it). Auditing is a highly desired feature. 

In order for auditing to work I would prefer to have the core-database in the same database as the tables I need to create and edit.

Now in order keep the database tidy I would be desirable to be able to create seperate schemas for bases created in nocodb.

**Reply by akrea** [reply]
BTW what would be the syntax to include the schema in the NC_DB env variable? I already use this command to start the container.
